doc/use_cli_in_airgapped_clouds.md

```diff
@@ -4,33 +4,66 @@
+```
+az extension add --source <local/path/to/the/extension/wheel>
+```
+Otherwise, you need to figure out the dependency tree of the packages in `run_requires` part using a tool like [pipdeptree](https://pypi.org/project/pipdeptree/). Download all the missing dependency packages that are not already [installed by Azure CLI](https://github.com/Azure/azure-cli/blob/master/src/azure-cli/requirements.py3.Linux.txt) from https://pypi.org/ and use `pip` to install the extension without dependencies in the default extension directory and then install all dependency packages locally. Take `azure-devops` as an example, it requires
+```
+"distro (==1.3.0)",
+"msrest (<0.7.0,>=0.6.0)",
+"python-dateutil (==2.7.3)"
+```
+
+We can first download `azure-devops` wheel from https://github.com/Azure/azure-devops-cli-extension/releases/download/20190805.1/azure_devops-0.12.0-py2.py3-none-any.whl after looking up in the [index.json](https://github.com/Azure/azure-cli-extensions/blob/master/src/index.json). For dependencies, a compatible version of `msrest` is already installed by Azure CLI as it is in the [requirements.txt](https://github.com/Azure/azure-cli/blob/master/src/azure-cli/requirements.py3.Linux.txt). `pipdeptree -p distro` shows that `distro` has no dependencies and we can download it from https://pypi.org/project/distro/1.3.0/#files. `pipdeptree -p python-dateutil` shows `python-dateutil` depends on `six` and `six` is already installed by Azure CLI. We only need to download `python-dateutil` from https://pypi.org/project/python-dateutil/2.7.3/#files.
+
+Therefore, `azure-devops` can be installed with the following commands (assuming packages are already downloaded in the home directory):
+
+```
+$ pip install --no-deps ~/azure_devops-0.12.0-py2.py3-none-any.whl --target ~/.azure/cliextensions/azure-devops
+$ pip install --no-deps ~/distro-1.3.0-py2.py3-none-any.whl --target ~/.azure/cliextensions/azure-devops
+$ pip install --no-deps ~/python_dateutil-2.7.3-py2.py3-none-any.whl --target ~/.azure/cliextensions/azure-devops
+```
 
 ## Load Cloud Endpoints
-If you are working in an Azure AirGapped Cloud, you should be able to get a cloud metadata URL from its documentation. You can set the environment variable `ARM_CLOUD_METADATA_URL` to this URL, then CLI will load the available clouds and the corresponding cloud endpoints from the URL. The first cloud in the available cloud list will be set as the active cloud by default if the public `AzureCloud` is (most likely) not available.
+If you are working in an Azure AirGapped Cloud, you should be able to get a cloud metadata URL from its documentation. You can set the environment variable `ARM_CLOUD_METADATA_URL` to this URL with:
+```bash
+export ARM_CLOUD_METADATA_URL=https://example.url.com/metadata/endpoints?api-version=2019-05-01
+```
+on Linux shells or
+```powershell
+$Env:ARM_CLOUD_METADATA_URL = "https://example.url.com/metadata/endpoints?api-version=2019-05-01"
+```
+on Windows Powershell.
```

Comment:
An example metadata file is https://management.azure.com/metadata/endpoints?api-version=2019-05-01